internal/providers/pluginfw/converters/tf_to_go.go

```diff
@@ -98,6 +100,26 @@ func tfSdkToGoSdkSingleField(ctx context.Context, srcField reflect.Value, destFi
 		if TfSdkToGoSdkStruct(ctx, srcFieldValue, destField.Interface()).HasError() {
 			panic(fmt.Sprintf("%s. %s", tfSdkToGoSdkStructConversionFailureMessage, common.TerraformBugErrorMessage))
 		}
+	} else if srcField.Kind() == reflect.Slice && destField.Kind() == reflect.Struct {
+		if srcField.IsNil() {
+			// Skip nils
+			return nil
+		}
+		assertStructSliceLengthIsOne(srcField)
+		tfsdkToGoSdkStructField(srcField.Index(0), destField, srcFieldName, forceSendFieldsField, ctx)
```

Comment:
For my own understanding, why can we call the internal method here but need to call the public method in the Ptr case?

Reply:
I think they are kind of equivalent because eventually the internal function will call the public method recursively